lib/index.js

```diff
@@ -177,6 +173,27 @@ exports.request = function (method, url, options, callback, _trace) {
         req.write(options.payload);
     }
 
+    // Custom abort method to detect early aborts
+
+    var _abort = req.abort;
+    var aborted = false;
+    req.abort = function () {
```

Comment:
Who else can call `abort()` other than line 87? Because is no one, then why override the node internals?